src/cluster.jl

```diff
@@ -99,10 +99,10 @@ mutable struct Worker
     del_msgs::Array{Any,1} # XXX: Could del_msgs and add_msgs be Channels?
     add_msgs::Array{Any,1}
     @atomic gcflag::Bool
-    state::WorkerState
-    c_state::Condition      # wait for state changes
-    ct_time::Float64        # creation time
-    conn_func::Any          # used to setup connections lazily
+    @atomic state::WorkerState
```

Comment:
if state is always read/written from inside a lock this doesn't need to be atomic.

Reply:
I don't think that's guaranteed? From a cursory grep through `cluster.jl` I see plenty of reads outside of a lock.